exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java

```diff
@@ -356,7 +358,25 @@ private void writeInt32(int readInt32, final MapOrListWriterImpl writer, String
     }
   }
 
-  public void ensureAtLeastOneField(ComplexWriter writer) {
+    private void writeDecimal128(BigDecimal readBigDecimal, final MapOrListWriterImpl writer, String fieldName, boolean isList) {
+        final VarDecimalHolder varDecimalHolder = new VarDecimalHolder();
+        byte[] bytes = readBigDecimal.unscaledValue().toByteArray();
+        ensure(bytes.length);
+        workBuf.setBytes(0, bytes);
+        varDecimalHolder.buffer = workBuf;
+        varDecimalHolder.scale = readBigDecimal.scale();
+        varDecimalHolder.precision = readBigDecimal.precision();
+        varDecimalHolder.start = 0;
+        varDecimalHolder.end =  bytes.length;
+        if (isList == false) {
+            writer.varDecimal(fieldName, varDecimalHolder.precision, varDecimalHolder.scale).write(varDecimalHolder);
+        } else {
+            writer.list.varDecimal().write(varDecimalHolder);
+        }
```

Comment:
Writers have `writeVarDecimal()` method that handles storying `BigDecimal` values. Please use that method:
```suggestion
    if (isList) {
      writer.list.varDecimal().writeVarDecimal(readBigDecimal);
    } else {
      writer.varDecimal(fieldName, readBigDecimal.precision(), readBigDecimal.scale()).writeVarDecimal(readBigDecimal);
    }
```

Reply:
Done the suggested changes.

Reply:
You haven't removed the code where creating varDecimalHolder, it is not required for now.

Reply:
Sorry, that was a miss from side. Now I have done it.